airflow/configuration.py

```diff
@@ -469,11 +469,22 @@ class AirflowConfigParser(ConfigParser):
     }
 
     def __init__(self, *args, **kwargs):
+        default_config = kwargs.pop('default_config', None)
         ConfigParser.__init__(self, *args, **kwargs)
-        self.read_string(parameterized_config(DEFAULT_CONFIG))
+        self.defaults = ConfigParser(*args, **kwargs)
+
+        AirflowConfigParser._read_string(
+            self.defaults, parameterized_config(DEFAULT_CONFIG))
+        if default_config:
+            AirflowConfigParser._read_string(
+                self.defaults, parameterized_config(default_config))
         self.is_validated = False
 
     def read_string(self, string, source='<string>'):
+        AirflowConfigParser._read_string(self, string, source)
+
+    @staticmethod
```

Comment:
Ah, I see.

I feel like we're doing some rather ugly and convoluted stuff to work around a lack of `read_string` in some versions of ConfigParser; what if we monkeypatched it in before doing any of this:

Up near the top of the file, right after the imports, we can do the following:

```python
if not hasattr(ConfigParser, 'read_string'):
    # work around earlier versions of ConfigParser that do not support `read_string`
    def cp_read_string(self, string, source='<string>'):
        self.readfp(six.StringIO(string), filename=source)
    ConfigParser.read_string = cp_read_string
```

Then we can remove all of this logic from `AirflowConfigParser` and use `self.read_string` or `self.defaults.read_string`.